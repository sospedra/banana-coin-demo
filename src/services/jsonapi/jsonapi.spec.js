/* global expect, describe, it */
const { createJAR, getPageFromQuery, createLinks } = require('./')

describe('service jsonapi', () => {
  it('should return a valid JSON API Response', () => {
    const response = createJAR('batman', { nanana: '!' }, 1337)

    expect(response).toMatchSnapshot()
    expect(response).toBeAnInstanceOf(Object)
  })

  it('should parse the page form a req.query', () => {
    const number = 10
    const size = 60
    const query = { page: { number: number.toString(), size }}

    expect(getPageFromQuery()).toEqual({ number: 0, size: 1 })
    expect(getPageFromQuery(query)).toEqual({ number, size })
  })

  it('should return a jsonapi links object', () => {
    const number = 2
    const req = {
      protocol: 'http',
      get: () => 'localhost',
      originalUrl: `/api/route?page[number]=${number}&page[size]=10&gran=canaria`
    }

    expect(createLinks(req, number, 10)).toMatchSnapshot()
  })
})
