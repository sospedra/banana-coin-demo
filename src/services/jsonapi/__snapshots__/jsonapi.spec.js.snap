// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`service jsonapi should return a jsonapi links object 1`] = `
Object {
  "first": "http://localhost/api/route?page[number]=0&page[size]=10&gran=canaria",
  "last": "http://localhost/api/route?page[number]=10&page[size]=10&gran=canaria",
  "next": "http://localhost/api/route?page[number]=3&page[size]=10&gran=canaria",
  "prev": "http://localhost/api/route?page[number]=1&page[size]=10&gran=canaria",
  "self": "http://localhost/api/route?page[number]=2&page[size]=10&gran=canaria",
}
`;

exports[`service jsonapi should return a valid JSON API Response 1`] = `
Object {
  "attributes": Object {
    "nanana": "!",
  },
  "id": 1337,
  "type": "batman",
}
`;
